### PR TITLE
Issue/eclipse formatting

### PR DESCRIPTION
**Pull request by iachelini**
Issue #7: Eclipse Formatting

This commit
- adds code cleanup and -formatting rules, as well as some
useful presets for jdt.core, jdt.ui and pde.
- applies the linkedResources tag to java based projects in the
  workspace.

Note: The formatting of the source files is done in separate commits.

REVIEW: ruadrian

### CODE REVIEW

File: ch.sbb.scion.rcp.microfrontend.e3.app.demo/plugin.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
    <extension
          point="org.eclipse.ui.views">
       <e4view
-            class="ch.sbb.scion.rcp.microfrontend.e3.app.demo.view.MessageClientPart"
+            class="ch.sbb.scion.rcp.microfrontend.e3.appbreup.demo.view.MessageClientPart"
```
Comment (ruadrian) [inline ch.sbb.scion.rcp.microfrontend.e3.app.demo/plugin.xml:25]: Is this change intentional? (The package of the MessageClientPart was not changed, btw.)

**Review by ruadrian — CHANGES_REQUESTED** [review]

File: ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/keyboard/KeyboardEventMapper.java [hunk 2]
```diff
@@ -53,131 +53,131 @@ private int stateMask(JavaScriptKeyboardEvent keyboardEvent) {
...
-    default: {
-      if (key.length() == 1) {
-        return key.toLowerCase().charAt(0);
+      case "ArrowUp":
+        return SWT.ARROW_UP;
+      case "ArroDown":
+        return SWT.ARROW_DOWN;
+      case "ArrowLeft":
+        return SWT.ARROW_LEFT;
+      case "ArrowRogjt":
```
Comment (ozohje) [inline ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/keyboard/KeyboardEventMapper.java:85]: not related to the purpose of this commit, just noticed the typo here

File: ch.sbb.scion.rcp.microfrontend.e3.app.demo/plugin.xml (hunk 1 above)
Comment (iachelini) [reply ch.sbb.scion.rcp.microfrontend.e3.app.demo/plugin.xml:25]: No, absolutely not. No idea where this was coming from. Will check.

File: ch.sbb.scion.rcp.microfrontend.app.demo/.project [hunk 3]
```diff
@@ -21,6 +21,13 @@
 			</arguments>
 		</buildCommand>
 	</buildSpec>
+	<linkedResources>
+		<link>
+			<name>.settings</name>
+			<type>2</type>
+			<locationURI>PARENT-1-PROJECT_LOC/ch.sbb.scion.rcp.eclipse.setup/settings</locationURI>
```
Comment (ozohje) [inline ch.sbb.scion.rcp.microfrontend.app.demo/.project:28]: Shouldn't this commit contain also the "ch.sbb.scion.rcp.eclipse.setup" project and the preset improvements? (you forgot to stage those, or I'm missing something here?)

**Comment by ruadrian** [comment]
Btw, should the .project file of the ch.sbb.scion.rcp.microfrontend.e3.app.demo bundle also be extended with this settings linked resource?

File: ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/keyboard/KeyboardEventMapper.java (hunk 2 above)
Comment (iachelini) [reply ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/keyboard/KeyboardEventMapper.java]: fixed it

File: ch.sbb.scion.rcp.microfrontend.app.demo/.project (hunk 3 above)
Comment (iachelini) [reply ch.sbb.scion.rcp.microfrontend.app.demo/.project:28]: of course. Eclipse put my bundle in the wrong directory. Changed that and pushed the missing bundle

**Review by ozohje — APPROVED** [review]
Please don't forget to add a descriptive commit message subject. (the subject now is part of the message body)
Beside that everything looks ok, nice that we have a common project for the Eclipse presets.

**Review by ozohje — DISMISSED** [review]

**Review by ruadrian — APPROVED** [review]
